### PR TITLE
Add doc of DynamoDB sink

### LINKED ISSUE #2228: Document: feat(connector): add DynamoDB sink
This issue tracks the documentation update needed for the merged PR #16670.

Source PR URL: https://github.com/risingwavelabs/risingwave/pull/16670
Source PR Merged At: 2024-06-04T04:30:20Z

### PR DESCRIPTION
**Pull request by WanYixian**
## Info

- **Description**

  - Add doc of DynamoDB sink

- **Related code PR**

  - https://github.com/risingwavelabs/risingwave/pull/16670

- **Related doc issue**
  
  Resolves https://github.com/risingwavelabs/risingwave-docs/issues/2228

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`emile-00`, `hengm3467`, & `WanYixian`).

### CODE REVIEW

File: docs/guides/sink-to-dynamodb.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+| `aws.credentials.secret_access_key`| Conditional. AWS secret access key for accessing DynamoDB. Must be specified when using access key ID and secret key.   |
+| `aws.credentials.role.arn`         | Conditional. ARN of the IAM role to assume for accessing DynamoDB. Must be specified when using AssumeRole.             |
+| `aws.credentials.role.external_id` | Conditional. External ID for assuming the IAM role specified in `aws.credentials.role.arn`.                              |
+| `aws.profile`                      | Optional. The name of the AWS CLI profile to use for accessing DynamoDB. If specified, it overrides the default profile. |
+| `dynamodb.max_batch_rows`          | Optional. Maximum number of rows to write in a single batch operation to DynamoDB. This helps optimize throughput and manage rate limits. Default value is `1024`.                      |
+
+## Partition key and sort key mapping
+
+For the DynamoDB sink to function correctly, it is required that the source table in RisingWave has a composite primary key comprising two columns, each corresponding to the partition key and sort key defined in the target DynamoDB table.
```
Comment (emile-00) [inline docs/guides/sink-to-dynamodb.md]:
```suggestion
For the DynamoDB sink to function correctly, it is required that the source table in RisingWave has a composite primary key comprising two columns, corresponding to the partition key and sort key defined in the target DynamoDB table.
```

**Review by emile-00 — APPROVED** [review]

**Review by ShanlanLi — APPROVED** [review]
LGTM

File: docs/guides/sink-to-dynamodb.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+| RisingWave Data Type        | DynamoDB Data Type |
+|-----------------------------|--------------------|
+| boolean                     | Bool               |
+| smallint                    | number (N)         |
+| integer                     | number (N)         |
+| bigint                      | number (N)         |
+| numeric                     | number (N)         |
+| real                        | number (N)         |
+| double precision            | number (N)         |
+| character varying (varchar) | string (S)         |
```
Comment (jetjinser) [inline docs/guides/sink-to-dynamodb.md]:
```suggestion
| double precision            | number (N)         |
| serial                      | number (N)         |
| character varying (varchar) | string (S)         |
```

**Review by jetjinser — CHANGES_REQUESTED** [review]
LGTM! Just supplement the datatype in a map table

**Review by jetjinser — APPROVED** [review]

**Pull request merged**